### ISSUE TITLE
Pure float32: every Cue model has a NaN objective gradient — not the exponent-ceiling defects, which are fixed

### ISSUE THREAD

**Issue by suchethac** [suchethac/tengri#1719]
The pure-float32 objective gradient is NaN for **every** model carrying
`neb={'type': 'cue'}`, while the forward pass is finite and float64 is clean. No
other physics block does this.

Split out of #1206 / #1388 because the two exponent-ceiling defects found while
chasing it (#1206, fixed in `84ce1b602`, `264b46413`, `6870673fe`) turned out
**not** to be the cause: they are fixed, verified, and this still reproduces.

## Measured

`cf2eb1b22`..`6870673fe`, CPU backend, `jax.enable_x64(False)`. One mock generated in
float64 and reused by both precisions; the gradient evaluated **away from the truth**
(at the truth the residual is exactly zero, the float64 gradient is legitimately
`[0,0]` and the comparison is vacuous — this cost me two probe runs).

`grad(chi2)` w.r.t. `sfh_delayed_log_total_mass` and `dust_tau_diff`, adding one
block at a time to stellar+dust:

| variant | pure-f32 gradient | finite | rel vs f64 |
|---|---|---|---|
| stellar + dust | `[-628.67, 6626.92]` | yes | 1.32e-05 |
| + dust IR (dale2014) | `[1476.57, 9310.53]` | yes | 1.23e-05 |
| **+ neb cue** | **`[nan nan]`** | **NO** | — |
| + agn composable | `[1476.57, 9310.53]` | yes | 1.23e-05 |
| + radio | `[1476.56, 9310.56]` | yes | 9.14e-06 |
| + xray | `[1476.56, 9310.56]` | yes | 8.82e-06 |
| + shock | `[1476.82, 9311.53]` | yes | 1.01e-05 |

On the panchromatic kitchen-sink model (`recipes.agn_panchromatic()`, `approx=None`)
the acceptance sweep fails at **every** corner — 0, ±3, ±6, ±10 z-scores in
standardized space — with **24 of 25** gradient components non-finite. The centre
fails too, so this needs no corner to reproduce.

The forward is also affected on the panchromatic model (`predict_photometry` non-finite
there), which is what poisons `L_absorbed` / `L_ir` / `log_L_ir` downstream. On the
minimal `+ neb cue` model above the forward is finite and only the gradient is NaN.

## What is ruled out

* **Both exponent-ceiling defects.** `10.0 ** jnp.clip(exponent, -50.0, 50.0)` in
  `predict_all_lines` / `predict_continuum` was `inf` in float32 (ceiling `10**38.53`),
  and the bound is now `representable_exponent(50.0)` — sized for the **derivative**,
  since `d/dx 10**x = 10**x·ln10` overflows at a value-sized cap. That fixed the
  pure-f32 Cue *forward* on the minimal model, neuter-checked. It did **not** fix this.
* **`_derive_cue_params_from_ssp`.** Already log-domain: `logsumexp` for `total_logqion`
  and the per-segment max-offset for `alpha_eff` / `logLratio_eff` (`cue.py:1449-1498`).
* **The qsogen `exp(clip(x, 0, 500))` family** (`264b46413`) — different component, and
  `+ agn composable` is clean above regardless.
* **The linear `nion` readers.** `line_precompute.py:80` and
  `nebular_grid_precompute.py:307` are precompute-only; every measurement here is
  `approx=None`.

## What is NOT localized

Which operation inside the Cue path produces the non-finite value. Two honest caveats
on the obvious next step:

1. Probing `predict_all_lines` / `predict_continuum` directly needs the real 12-wide
   `nn_params`. Synthetic values (e.g. `[-2.5, 2.0, -0.3, 0, 0, ...]`) **saturate the
   clip at every `gas_logqion` in both precisions** — measured `lum max` pinned at
   exactly `10**50` (f64) and the f32 cap — so such a probe is not representative of
   the model path and its verdicts should not be trusted. Drive the functions with the
   `nn_params` the component actually assembles.
2. Forward-value comparison is the wrong instrument here. The sibling defect in
   `264b46413` had a **bit-identical** forward value with and without the bug and
   differed only in the gradient, because `1/(exp(inf)-1)` is `0.0` — the correct
   Planck tail. Instrument the reverse pass.

## Reproduce

```python
import jax, jax.numpy as jnp, numpy as np
from tengri import FIXED, Fixed, Observation, Photometry, SEDModel, Uniform
from tengri.components.stellar.sps.dsps_wrapper import load_ssp_data

TRUTH = {"sfh_delayed_log_total_mass": 10.0, "dust_tau_diff": 0.5}
EVAL  = {"sfh_delayed_log_total_mass": 10.3, "dust_tau_diff": 0.7}   # NOT the truth

def build(ssp, neb):
    kw = {} if neb is None else {"neb": {"type": neb, "all_params": FIXED}}
    return SEDModel.build(
        ssp_data=ssp,
        observation=Observation(photometry=Photometry.from_names(
            ["galex_fuv", "sdss_r", "wise_w1", "wise_w4", "herschel_250"])),
        sfh={"type": "delayed", "all_params": FIXED,
             "log_total_mass": Uniform(9.0, 11.0), "tau_gyr": 1.0, "age_gyr": 5.0},
        dust={"type": "two_component", "law_bc": "calzetti", "all_params": FIXED,
              "tau_diff": Uniform(0.0, 1.5), "tau_bc": 0.0,
              "emission": {"type": "dale2014", "all_params": FIXED}},
        redshift=Fixed(0.1), approx=None, **kw)

def grad(neb, x64, dt, data, noise):
    with jax.enable_x64(x64):
        model = build(load_ssp_data("data/fsps_prsc_miles_chabrier.h5"), neb)
        d, s = jnp.asarray(data, dt), jnp.asarray(noise, dt)
        names = sorted(TRUTH)
        def nlp(v):
            r = (model.predict_photometry({k: v[i] for i, k in enumerate(names)}) - d) / s
            return 0.5 * jnp.sum(r * r)
        return jax.grad(nlp)([jnp.asarray(EVAL[k], dt) for k in names])

with jax.enable_x64(True):
    m = build(load_ssp_data("data/fsps_prsc_miles_chabrier.h5"), "cue")
    flux = np.asarray(m.predict_photometry(TRUTH), np.float64)

print("f64:", grad("cue", True,  jnp.float64, flux, np.abs(flux) * 0.05))
print("f32:", grad("cue", False, jnp.float32, flux, np.abs(flux) * 0.05))  # -> [nan nan]
```

## Why it matters

This is what gates #1206's acceptance criterion — *"the full panchromatic forward model
runs finite and accurate under `jax.enable_x64(False)` end-to-end"*. Cue is the default
nebular backend for bare-stellar SSPs, so in practice it blocks pure-float32 inference
for realistic models regardless of what else is fixed.

Related: #1206, #1388, #1534.

**Comment by suchethac** [comment]
Determination after systematic re-testing on current main (post the merged float32 work: #1868's import/tolerance blockers, #1859's representable flux scales, the earlier #1206 exponent-ceiling fixes): **the pure-float32 NaN objective gradient does not reproduce.**

Measured (PR #1905 carries the regression tests):
- Minimal recipe from this issue (delayed SFH + two_component dust + neb cue, mock reused across precisions, gradient away from truth): **13/13 pure-f32 gradients finite** through the production whitened chi2.
- `recipes.agn_panchromatic()`, `approx=None`, standardized center: forward photometry **finite** in pure f32; **20/25 gradients finite** — the residual five are `agn_{oa,p,q,tau}_skirtor` + `redshift`, i.e. the known-open #1860 family (filter-integral VJP) and SKIRTOR geometry, not the nebular path. That is a large improvement over this issue's measurement (24/25 non-finite AND non-finite forward).
- Methodological caveat that cost this investigation two rounds and is now pinned in the test file: a hand-rolled `sum(r**2/err**2)` chi2 at these flux scales (~1e-28) **underflows to 0/0 = NaN in f32 by itself** — the production ratio-first whitening (`((pred-obs)/err)**2`) is the only valid form, and a NaN measured through the naive form is the probe's artifact, not the model's.

The double-where guards on the two cue exponent sites land in PR #1905 as hardening (pinned bit-identical on finite f64 inputs), together with `tools/check_numeric_guards.py` — the CI census that keeps this defect class (raw floors / clip-then-power outside `utils/scale.py`) from growing back.

Closing as not-reproducible-on-main with regression guards in place; the residual skirtor/redshift f32 gradient NaNs are tracked by #1860 and its follow-ups.
